inbm/telemetry-agent/telemetry/telemetry_handling.py

```diff
@@ -303,7 +303,7 @@ def get_query_related_info(option: str, info: Dict) -> Dict:
                 del info['values'][elem]
     elif option == "version":
         info = {}
-        info['INBMVersionCommit'] = get_friendly_inbm_version_commit()        
+        info['INBMVersionCommit'] = get_friendly_inbm_version_commit()
     elif option == "all":
```

Comment:
**<h2>$\color{#FFA500}Improvements: CodeReliability$</h2>**
**Suggestion:** Replace the direct manipulation of sleep time counters with a more robust timing mechanism, such as using time.monotonic() or a dedicated timer class to handle sleep intervals and conditions.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                # Example improvement using time.monotonic()
                current_time = time.monotonic()
                if current_time - self.__last_publish_time >= self.__publish_time:
                    self.__last_publish_time = current_time
                    self.__publish_counter += 1
                ...
                if current_time - self.__last_collect_time >= self.__collect_time:
                    self.__last_collect_time = current_time
                    self.__collect_counter += 1
```

Reply:
good suggestion but out of scope